libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -32,6 +32,41 @@ using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace
+{
+
+void copyMissingTags(StructurallyDocumentedAnnotation& _target, CallableDeclarationAnnotation const& _source)
+{
+	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>(_source);
+
+	set<string> existingTags;
+
+	for (auto const& iterator: _target.docTags)
+		existingTags.insert(iterator.first);
+
+	for (auto const& [tag, content]: sourceDoc.docTags)
+		if (tag != "inheritdoc" && !existingTags.count(tag))
+			_target.docTags.emplace(tag, content);
+}
+
+CallableDeclaration const* findBaseCallable(std::set<CallableDeclaration const*>& _baseFuncs, string const& _name)
+{
+	for (CallableDeclaration const* baseFuncCandidate: _baseFuncs)
+		if (baseFuncCandidate->annotation().contract->name() == _name)
```

Comment:
If we do it properly, this should do actual reference resolution - would that be very complicated?
The reason I'm bringing this up is that this name might not be the name under which the contract is known in the current file.